ts/src/cryptocom.ts

```diff
@@ -2651,6 +2652,93 @@ export default class cryptocom extends Exchange {
         };
     }
 
+    async fetchSettlementHistory (symbol: string = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name cryptocom#fetchSettlementHistory
+         * @description fetches historical settlement records
+         * @see https://exchange-docs.crypto.com/exchange/v1/rest-ws/index.html#public-get-expired-settlement-price
+         * @param {string} symbol unified market symbol of the settlement history
+         * @param {int|undefined} since timestamp in ms
+         * @param {int|undefined} limit number of records
+         * @param {object} params exchange specific params
+         * @param {int|undefined} params.type 'FUTURE', 'WARRANT'
+         * @returns {[object]} a list of [settlement history objects]
+         */
+        const type = this.safeStringUpper2 (params, 'type', 'instrument_type');
+        this.checkRequiredArgument ('fetchSettlementHistory', type, 'type', [ 'FUTURE', 'WARRANT' ]);
```

Comment:
@Dan-krm , shouldn't we use the unified terms here? In that case couldn't we be using `handleMarketTypeAndParams` so it fallbacks to `defaultType` if the user does not provide `type` in `params`?

Reply:
Yeah I'll adjust to use handleMarketTypeAndParams